src/transformers/models/gpt_neox/modeling_gpt_neox.py

```diff
@@ -297,21 +297,22 @@ def __init__(self, dim, max_position_embeddings, base=10000, device=None):
         self.dim = dim
         self.max_position_embeddings = max_position_embeddings
         self.base = base
-        inv_freq = 1.0 / (self.base ** (torch.arange(0, self.dim, 2).float().to(device) / self.dim))
```

Comment:
float() always turns a tensor to float32.   This is why initialisation with dtype=float16 did'n work.